src/o365/spfx/commands/project/project-upgrade.ts

```diff
@@ -488,6 +507,99 @@ ${f.resolution}
     return s.join('').trim();
   }
 
+  private getTourReport(findings: FindingToReport[]): string {
+    const tourFindings: FindingTour = {
+      title: "Upgrade Project",
+      steps: []
+    };
+
+    let addedBuildStep: boolean = false;
+    const buildStep: FindingTourStep = {
+      file: "./package.json",
+      title: "REQUIRED: Build project",
+      description: "### REQUIRED\r\n\r\nBuild the project\r\n>> gulp build",
+      line: 1
+    };
+
+    findings.forEach(f => {
+
+      const lineNumber: number = f.position && f.position.line ? f.position.line : this.getLineToModify(f);
+      
+      let resolution: string = '';
+      switch (f.resolutionType) {
+        case 'cmd':
+          resolution = `Execute the following command:\r\n>> ${f.resolution}`;
+          break;
+        case 'json':
+        case 'js':
+        case 'ts':
+        case 'scss':
+          resolution = `Update the code as follows:\r\n\`\`\`${f.resolutionType}\r\n${f.resolution}\r\n\`\`\``;
+          break;
+      }
+
+      // If there is an NPM dedupe rule, instruct user to build
```

Comment:
Why should users build project after running `npm dedupe`?

Reply:
I try to instruct users to build before the dedupe step,  partly because of what we say in the dedupe instruction.

It is totally unnecessary, but I wanted to make sure that we left them with a project that builds ... so that if they introduce something that breaks the build after the upgrade, they don't blame the upgrade